### PR TITLE
Update plugin to support new HostedExperience interface

### PR DESCRIPTION
**Pull request by ngchphuong**
Update plugin to support new HostedExperience interface

### CODE REVIEW

File: spiff-connect/spiff-connect.php [hunk 1]
```diff
@@ -2,18 +2,21 @@
...
+Version: 2.0.67
 Description: Connect your WooCommerce store to Spiff and allow customers access to Spiff Workflows to personalize their products.
 Author: Spiff Pty. Ltd.
 License: GPL3
 */
 
 require plugin_dir_path(__FILE__) . 'includes/spiff-connect-requests.php';
 
-define("SPIFF_API_BASE", getenv("SPIFF_API_BASE"));
+define("SPIFF_API_BASE", "https://api.spiff.com.au");
```
Comment (elPixelador) [inline spiff-connect/spiff-connect.php]: Why has this been changed to be hardcoded?

File: spiff-connect/spiff-connect.php [hunk 2]
```diff
@@ -233,9 +236,17 @@ function spiff_enqueue_ecommerce_client() {
...
+        'spiff-assets-client',
+        SPIFF_API_ASSETS . 'api.js',
+        array(),
+        null
+    );
+
     wp_enqueue_script(
         'spiff-create-design-button',
-        plugin_dir_url(__FILE__) . 'public/js/create-design-button.js',
+        plugin_dir_url(__FILE__) . 'public/js/create-design-button.js'.'?'.rand(),
```
Comment (elPixelador) [inline spiff-connect/spiff-connect.php]: What is the purpose of adding rand here?

File: spiff-connect/public/js/create-design-button.js [hunk 3]
```diff
@@ -71,7 +100,9 @@ const executeTransaction = async (currencyCode, integrationProduct, wooProductId
   });
 
   transaction.execute();
-};
+}
 
 window.spiffAppendCreateDesignButton = spiffAppendCreateDesignButton;
-window.spiffAppendCreateDesignButtonBulk = spiffAppendCreateDesignButtonBulk;
+//window.spiffAppendCreateDesignButtonBulk = spiffAppendCreateDesignButtonBulk;
```
Comment (elPixelador) [inline spiff-connect/public/js/create-design-button.js]: Why has bulk mode been commented out?

File: spiff-connect/spiff-connect.php (hunk 1 above)
Comment (elPixelador) [reply spiff-connect/spiff-connect.php]: During the build process for woocommerce we place the file here.

spiff-connect/public/js/api.js

You can add it hear manually to see it working using that path

**Review by elPixelador — CHANGES_REQUESTED** [review]

File: spiff-connect/spiff-connect.php (hunk 2 above)
Comment (ngchphuong) [reply spiff-connect/spiff-connect.php]: Sorry, let me remove it, just use it to force cache

File: spiff-connect/public/js/create-design-button.js (hunk 3 above)
Comment (ngchphuong) [reply spiff-connect/public/js/create-design-button.js]: Just for debug something, we will revert it. Thank you!

File: spiff-connect/public/js/create-design-button.js [hunk 4]
```diff
@@ -59,21 +59,22 @@ const createButton = (text, buttonConfig, buttonClass) => {
...
 
-      button.onclick = () => executeTransaction(currencyCode, integrationProduct, wooProductId, redirectUrl, true);
+      button.onclick = () => showSpiffTransaction(product, currencyCode, integrationProductId, wooProductId, redirectUrl, true);
+      //button.onclick = () => executeTransaction(currencyCode, integrationProduct, wooProductId, redirectUrl, true);
     });
   });
 
-  integrationProduct.on('invalid', () => console.error("Spiff product could not be found."));
-  integrationProduct.confirmActive();
+  //integrationProduct.on('invalid', () => console.error("Spiff product could not be found."));
```
Comment (elPixelador) [inline spiff-connect/public/js/create-design-button.js]: Why has this been commented?

**Review by elPixelador — APPROVED** [review]

**Review by elPixelador — APPROVED** [review]

**Pull request merged**